runtime/binding-openapi-asyncapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/asyncapi/internal/config/composite/OpenapiAsyncapiCompositeGenerator.java

```diff
@@ -0,0 +1,236 @@
+import io.aklivity.zilla.runtime.engine.config.BindingConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.GuardedConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfigBuilder;
+
+public abstract class OpenapiAsyncapiCompositeGenerator
+{
+    public final OpenapiAsyncapiCompositeConfig generate(
+        OpenapiAsyncapiBindingConfig binding)
+    {
+        int tagIndex = 1;
+
+        final OpenapiParser openapiParser = new OpenapiParser();
+        final List<OpenapiSchemaConfig> openapiSchemas = new ArrayList<>();
+        for (OpenapiSpecificationConfig openapiSpec : binding.options.specs.openapi)
+        {
+            final String label = openapiSpec.label;
+
+            for (OpenapiCatalogConfig catalog : openapiSpec.catalogs)
+            {
+                final long catalogId = binding.resolveId.applyAsLong(catalog.name);
+                final CatalogHandler handler = binding.supplyCatalog.apply(catalogId);
+                final int schemaId = handler.resolve(catalog.subject, catalog.version);
+                final String payload = handler.resolve(schemaId);
+                final List<OpenapiServerConfig> configs =
+                    openapiSpec.servers == null || openapiSpec.servers.isEmpty()
+                        ? List.of(OpenapiServerConfig.builder().build())
+                        : openapiSpec.servers;
+                final OpenapiView openapi = OpenapiView.of(tagIndex++, label, openapiParser.parse(payload), configs);
+
+                openapiSchemas.add(new OpenapiSchemaConfig(label, schemaId, openapi));
+            }
+        }
+
+        final AsyncapiParser asyncapiParser = new AsyncapiParser();
+        final List<AsyncapiSchemaConfig> asyncapiSchemas = new ArrayList<>();
+        for (AsyncapiSpecificationConfig asyncapiSpec : binding.options.specs.asyncapi)
+        {
+            final String label = asyncapiSpec.label;
+
+            for (AsyncapiCatalogConfig catalog : asyncapiSpec.catalogs)
+            {
+                final long catalogId = binding.resolveId.applyAsLong(catalog.name);
+                final CatalogHandler handler = binding.supplyCatalog.apply(catalogId);
+                final int schemaId = handler.resolve(catalog.subject, catalog.version);
+                final String payload = handler.resolve(schemaId);
+                final List<AsyncapiServerConfig> configs =
+                    asyncapiSpec.servers == null || asyncapiSpec.servers.isEmpty()
+                        ? List.of(AsyncapiServerConfig.builder().build())
+                        : asyncapiSpec.servers;
+                final AsyncapiView asyncapi = AsyncapiView.of(tagIndex++, label, asyncapiParser.parse(payload), configs);
+
+                asyncapiSchemas.add(new AsyncapiSchemaConfig(label, schemaId, asyncapi));
+            }
+        }
+
+        return generate(binding, openapiSchemas, asyncapiSchemas);
+    }
+
+    protected abstract OpenapiAsyncapiCompositeConfig generate(
```

Comment:
Perhaps it would be better methods are sorted by access level. LIke all public goes first then protected.

Reply:
Yes, we are doing that, first for methods, then for inner classes / interfaces.